server/controllers/socket.controller.ts

```diff
@@ -50,34 +54,43 @@ async function handleJoin(userId: string, data: JoinEvent): Promise<boolean> {
 	if (!roomData) {
 		throw new Error(`No room with room code ${data.roomCode}.`);
 	}
-
+	let updatedRoomData;
 	const isModerator = !roomData.moderator;
 
-	const updatedRoomData = await rooms.findAndModify(
-		{ _id: roomData._id },
-		{
-			update: !isModerator
-				? {
-						$push: {
-							voters: {
-								$each: [
-									{
-										id: userId,
-										name: data.name,
-										confidence: null,
-										selection: null,
-									},
-								],
+	const timeoutIndex = timeouts.findIndex((item) => item.userId === userId);
+	if (timeoutIndex > -1) {
+		console.debug(
+			`User ${userId} rejoined room, cancelling deletion with timeoutId of ${timeouts[timeoutIndex].timeoutId}`,
+		);
+		clearTimeout(timeouts[timeoutIndex].timeoutId);
+		timeouts.splice(timeoutIndex, 1);
+		updatedRoomData = await rooms.findOne({ _id: roomData._id });
```

Comment:
Do we need to send a room update if nothing changed?

Reply:
Yes, or else the user who reconnects doesn't get the info. There may be a better way to do this, but I haven't found it.